### PR TITLE
The overview page can now display a different app than the default Rich Text Editor

### PR DESCRIPTION
**Pull request by claudiosousa**

### CODE REVIEW

**Comment by claudiosousa** [comment]
Hi @oliamb, here's the pull request to enable users ability to specify an 'app' name on the overview task to use that app as the overview of their collab in place of the default Rich Text App

File: src/automator/tasks/overview.js [hunk 1]
```diff
@@ -1,66 +1,91 @@
...
+     * If an 'entity' is specified, it will use the content of that storage file
+     * If an 'app' name is specified, it will use that app for the overview page
+     *
+     * The collab is indicated either by an id in `descriptor.collab` or a
+     * collab object in `context.collab`.
+     *
+     * @memberof hbpCollaboratory.hbpCollaboratoryAutomator.Tasks
+     * @param {object} descriptor the task configuration
+     * @param {object} [descriptor.collab] id of the collab
+     * @param {string} descriptor.entity either a label that can be found in
```
Comment (oliamb) [inline src/automator/tasks/overview.js]: You can now use the [] syntax which means the attribute is optional. [descriptor.entity] and [descriptor.app]

File: src/automator/tasks/overview.js [hunk 2]
```diff
@@ -1,66 +1,91 @@
...
+        return fetchSourceContent(descriptor, context)
+          .then(function(source) {
+            return $http.post(bbpConfig.get('api.richtext.v0') + '/richtext/', {
+              ctx: overview.context,
+              raw: source
+            });
+          });
+      };
+
+      var updateAppId = function(overview, descriptor) {
```
Comment (oliamb) [inline src/automator/tasks/overview.js]: You find the best way to do it, congratulations!

File: src/automator/tasks/overview.spec.js [hunk 3]
```diff
@@ -69,6 +77,7 @@ describe('overview task handler', function() {
       collab: data.collab.id,
       entity: data.fileEntity._uuid
     }).then(function(res) {
+      console.log('res:' + JSON.stringify(res));
```
Comment (oliamb) [inline src/automator/tasks/overview.spec.js]: No console.log please :)

File: src/automator/tasks/overview.spec.js [hunk 4]
```diff
@@ -80,16 +89,47 @@ describe('overview task handler', function() {
...
+      }).then(function(res) {
+        expect(res).toEqual(data.expectedNavItem);
+      });
+    scope.$digest();
+  }));
+
+  it('should update content appId', inject(function() {
+    spyOn(navStore, 'getRoot')
+      .and.returnValue(q.when(data.rootNavItem));
+    spyOn(fileStore, 'getContent')
```
Comment (oliamb) [inline src/automator/tasks/overview.spec.js]: Is this still called? I don't understand why you need to call getContent when you just want to update the navItem.

File: src/automator/tasks/overview.spec.js [hunk 5]
```diff
@@ -80,16 +89,47 @@ describe('overview task handler', function() {
...
+      .and.returnValue(q.when(data.fileContent));
+    backend.expectGET('http://collab/v0/extension/?title=My+app+name')
+      .respond(200, data.appResponse);
+    backend.expectPUT('http://collab/v0/collab/1/nav/10/').respond(201, {});
```
Comment (oliamb) [inline src/automator/tasks/overview.spec.js]: Minor, but on an update request the status code should be 200 OK, not 201 CREATED

**Comment by oliamb** [comment]
I made a few comments. You can stack new commits to your branch and they will be automatically added to this pull request.

Thanks for contributing!

**Comment by claudiosousa** [comment]
Applied the comments. hope I didn't forget anything

**Comment by oliamb** [comment]
Ok LGTM

**Comment by oliamb** [comment]
Version 0.1.5 contains your changes!

**Pull request merged**